### PR TITLE
Dmsgget implementation.

### LINKED ISSUE #101: Implement dmsgget.
**Feature description**

> TODO

### PR DESCRIPTION
**Pull request by evanlinjin**
Fixes #101

**Dmsgget implementation (Like wget, but over dmsg).**
This is one in a series of additions for the CX tracker to function.

Addtional changes:
* `dmsghttp.HTTPTransport` implementation now no longer closes the internal stream after .RoundTrip returns.

Notes:
* I implemented `dmsgget.FlagGroup` so that we don't need to duplicate code between the `main` package and `dmsgget` package. In the future, `dmsgget` will get a magnitude of different fields/flags.

How to test this PR:
* `make test`

### CODE REVIEW

**Comment by evanlinjin** [comment]
> A README would be appreciated for this as well as dmsghttp about basic usage.

I'll write an example for `dmsghttp`, but I don't think we need a README for `dmsgget`. One can just go: `dmsgget --help`.

File: dmsgget/dmsgget.go [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+func New(fs *flag.FlagSet) *DmsgGet {
+	dg := &DmsgGet{fs: fs}
+
+	for _, fg := range dg.flagGroups() {
+		fg.Init(fs)
+	}
+
+	w := fs.Output()
+	flag.Usage = func() {
+		_, _ = fmt.Fprintf(w, "Skycoin %s %s, wget over dmsg.\n", ExecName, Version)
```
Comment (nkryuchkov) [inline dmsgget/dmsgget.go:44]: Does there exist a possibility of an error returned by `fmt.Fprintf`?

File: dmsgget/dmsgget.go [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+	}
+
+	return errors.New("all download attempts failed")
+}
+
+func parseKeyPair(skStr string) (pk cipher.PubKey, sk cipher.SecKey, err error) {
+	if skStr == "" {
+		pk, sk = cipher.GenerateKeyPair()
+		return
+	}
```
Comment (nkryuchkov) [inline dmsgget/dmsgget.go:142]: Could you please add a new line after this `if` statement and after one below?

File: dmsgget/dmsgget.go [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+	}
+	pk, err = sk.PubKey()
+	return
+}
+
+func parseURL(args []string) (*URL, error) {
+	if len(args) == 0 {
+		return nil, ErrNoURLs
+	}
```
Comment (nkryuchkov) [inline dmsgget/dmsgget.go:154]: Could you please add new lines after `if` statements in this function?

File: dmsgget/flags.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
+package dmsgget
+
+import "flag"
+
+// Constants related to dmsgget
+const (
+	ExecName = "dmsgget"
+	Version  = "v0.1.0"
```
Comment (nkryuchkov) [inline dmsgget/flags.go:8]: Would it be more convenient to take a version from `git describe`?

File: dmsgget/url.go [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+// URL represents a dmsg http URL.
+type URL struct {
+	dmsg.Addr
+	url.URL
+}
+
+// Fill fills the internal fields from an URL string.
+func (du *URL) Fill(str string) error {
+	u, err := url.Parse(str)
+	if err != nil {
```
Comment (nkryuchkov) [inline dmsgget/url.go:25]: Could you please add new lines after `if` statements?

File: dmsgget/dmsgget.go (hunk 1 above)
Comment (evanlinjin) [reply dmsgget/dmsgget.go:44]: > Does there exist a possibility of an error returned by `fmt.Fprintf`?

@nkryuchkov It is writing on either `stderr` or `stdout` so I think not.

**Pull request merged**